### PR TITLE
Fixed the bugs in chat Activity

### LINKED ISSUE #7: Notification Integration
Integration of firebase cloud messaging in the app to notify users of messages.

### LINKED ISSUE #88: Fix bugs in ChatActivity
There are 2 bugs that I have discovered -
1 number of chats in the chat fragment tab heading are increasing even if the messages are from the same person
2 if you clear the whole chat, then also the last message is visible in the chat list item

### PR DESCRIPTION
**Pull request by riturajjain2000**
Fixes #7 #88 #91 #92 #98 #99

### CODE REVIEW

**Comment by plazzy99** [comment]
raise all your pr's i will review it later

**Comment by riturajjain2000** [comment]
> raise all your pr's i will review it later

OK

**Comment by plazzy99** [comment]
Please mention in a proper documented message which files are connected with which pr

**Comment by riturajjain2000** [comment]
> Please mention in a proper documented message which files are connected with which pr

ok will do that shortly

**Comment by riturajjain2000** [comment]
Issue #99 - Commit "Fixed Crahing and visibilty of Image as Last message" & 3 files changed
Issue #98 - Commit "Added Animation to Splash Screen screen" & 2 files changed
Issue #92 - Commit "Added Animations to buttons" & 18 files changed
Issue #91 - Commit "Login Screen UI" & 5 files changed
Issue #88 - Commit "Fixed bugs" & 2 files changed
Issue #7- Commit "Notifications Integration" & 15 files changed

File: app/src/main/java/com/vatsal/kesarwani/therapy/Activity/ChatActivity.java [hunk 1]
```diff
@@ -647,6 +663,62 @@ private void post() {
...
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+
+            }
+        });
+    }
+
+    private void sendNotification(String receiver, final String name, final String msg) {
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Activity/ChatActivity.java:686]: notification issue solved 👍🏻

Comment (plazzy99) [reply app/src/main/java/com/vatsal/kesarwani/therapy/Activity/ChatActivity.java:686]: #7

File: app/src/main/java/com/vatsal/kesarwani/therapy/Activity/AddPost.java [hunk 2]
```diff
@@ -136,6 +138,8 @@ public void onClick(View v) {
                     return;
                 }
                 post.setEnabled(false);
+                Animation animation = AnimationUtils.loadAnimation(AddPost.this,R.anim.blink_anim);
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Activity/AddPost.java:141]: Added animated button #92  solved 👍🏻

File: app/src/main/java/com/vatsal/kesarwani/therapy/Activity/Editprofile.java [hunk 3]
```diff
@@ -77,6 +79,8 @@ protected void onCreate(Bundle savedInstanceState) {
         save.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                Animation animation = AnimationUtils.loadAnimation(Editprofile.this,R.anim.blink_anim);
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Activity/Editprofile.java:82]: #92 👍🏻

File: app/src/main/java/com/vatsal/kesarwani/therapy/Activity/LoginScreen.java [hunk 4]
```diff
@@ -40,9 +43,9 @@
 public class LoginScreen extends AppCompatActivity {
 
     private static final int RC_SIGN_IN = 10;
-    private Button google;
+    private SignInButtonImpl google;
     private FirebaseAuth mAuth;
-    private GoogleSignInClient mGoogleSignInClient ;
+    private GoogleSignInClient mGoogleSignInClient;
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Activity/LoginScreen.java:48]: Login screen UI #91

Comment (plazzy99) [reply app/src/main/java/com/vatsal/kesarwani/therapy/Activity/LoginScreen.java:48]: 👍🏻

File: app/src/main/java/com/vatsal/kesarwani/therapy/Fragment/APIService.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import com.vatsal.kesarwani.therapy.Notifications.Sender;
+
+import retrofit2.Call;
+import retrofit2.http.Body;
+import retrofit2.http.Headers;
+import retrofit2.http.POST;
+
+public interface APIService {
+    @Headers(
+            {
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Fragment/APIService.java:13]: notification

File: app/src/main/res/layout/activity_splash.xml [hunk 6]
```diff
@@ -20,8 +20,24 @@
...
         android:textStyle="bold"
         app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.361" />
+
+    <com.airbnb.lottie.LottieAnimationView
```
Comment (plazzy99) [inline app/src/main/res/layout/activity_splash.xml:29]: #98  👍🏻

File: app/src/main/java/com/vatsal/kesarwani/therapy/Adapter/ChatAdapter.java [hunk 7]
```diff
@@ -230,76 +220,85 @@ public int getItemCount() {
...
-                {try
-                {
-                    MessageModel messageModel =dataSnapshot.getValue(MessageModel.class);
-                    if (messageModel.getReceiver().equals(firebaseUser.getUid()) && messageModel.getSender().equals(userid) || messageModel.getReceiver().equals(userid) && messageModel.getSender().equals(firebaseUser.getUid()) ){
-                        lastmssg=messageModel.getMssg();
-                        
+                for (DataSnapshot dataSnapshot : snapshot.getChildren()) {
+
+                    MessageModel messageModel = dataSnapshot.getValue(MessageModel.class);
+                    if (messageModel.getReceiver().equals(firebaseUser.getUid()) && messageModel.getSender().equals(userid) || messageModel.getReceiver().equals(userid) && messageModel.getSender().equals(firebaseUser.getUid())) {
```
Comment (plazzy99) [inline app/src/main/java/com/vatsal/kesarwani/therapy/Adapter/ChatAdapter.java:269]: #88 👍🏻

**Review by plazzy99 — APPROVED** [review]

**Comment by plazzy99** [comment]
Issue #99 please explain this implementation. Mentioning the file name @riturajjain2000

**Comment by riturajjain2000** [comment]
> Issue #99 no explicit code found for its resolution thus points will not be awarded for this issue 👎🏻

The Files namely ChatActivity and ChatAdpater were changed to resolve this issue only, the commit is named as - "Fixed Crahing and visibilty of Image as Last message"

**Comment by plazzy99** [comment]
@riturajjain2000 got it

**Comment by plazzy99** [comment]
giving you 2+10 points

**Comment by riturajjain2000** [comment]
> giving you 2+10 points

Thanks !

**Pull request merged**